src/Build/Instance/TaskRegistry.cs

```diff
@@ -1790,10 +1766,12 @@ public void Translate(ITranslator translator)
 
             IDictionary<RegisteredTaskIdentity, List<RegisteredTaskRecord>> copy = _taskRegistrations;
             translator.TranslateDictionary(ref copy, TranslateTaskRegistrationKey, TranslateTaskRegistrationValue, count => CreateRegisteredTaskDictionary(count));
+            // Ensure that mutations of the deserialized task registry are getting unique order ids.
+            s_nextRegistrationOrderId = Math.Max(s_nextRegistrationOrderId, copy.Count);
```

Comment:
`copy.Count` may be smaller than the maximum order ID on the serializing side. Shouldn't you actually make `s_nextRegistrationOrderId` part of the payload?

Reply:
That should be fine - we only need to prevent colisions. 

If receiving site already have the `s_nextRegistrationOrderId` greater than what was the last order id on sending side, than it will not have collisions after possible mutations.

The only possible catch would be if we'd allow transfering registeret tasks between different instances of TaskRegistry on different nodes - in such case we'd need to maintain some form of globla distributed identity (e.g. [snowflake ids](https://github.com/RobThree/IdGen)) - but that is overkill now